File: showcases/dancing-skeleton.ts
// Browser showcase for the animated dancing-skeleton glTF asset.
//
// The reusable ObjViewer component fetches scene.gltf and its external
// scene.bin buffer; this module only describes which animation to play.
export async function dancingSkeleton(viewer: any) {
  // Loading returns only after the scene has been parsed and applied to the
  // active viewer, so animation commands below cannot race scene creation.
  const animations = await viewer.loadGltf('/models/dancing-skeleton/scene.gltf')

  // Keep the camera stable and frame the imported mesh before playback starts.
  viewer.resetCamera()

  // The asset currently exposes its Mixamo dance clip as animation zero.
  if (animations.length > 0) {
    viewer.selectAnimation(0)
    viewer.setAnimationSpeed(1.0)
    viewer.playAnimation()
  }
}

File: scripts/previews/dancing-skeleton.lua
set_background_color({ 1, 1, 1 })
local animations = load_gltf("../public/models/dancing-skeleton/scene.gltf")
reset_camera()
if #animations > 0 then
  select_animation(1)
  set_animation_time(0.4)
  set_motion_lines_random(256, 15)
end
show_scene()
save_screenshot("../public/previews/dancing-skeleton.png")
